### ISSUE TITLE
Misleading documentation about the supported OIDC signing algorithms 

### ISSUE THREAD

**Issue by vpnachev** [kubernetes/kubernetes#104073]
#### What happened:
A kube-apiserver does not support all of the signing algorithms listed in RFC 7518 https://tools.ietf.org/html/rfc7518#section-3.1.
It fails with something like 
```
Error: oidc: unsupported signing alg: "HS256"
```

When looking into the [documentation](https://v1-20.docs.kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/) or in the code https://github.com/kubernetes/kubernetes/blob/5be21c50c269fc1d28e0bd31ab9dcb572ae7fac5/pkg/kubeapiserver/options/authentication.go#L300-L303

I really get the impression that all algorithms from the RFC document are supported
```
   +--------------+-------------------------------+--------------------+
   | "alg" Param  | Digital Signature or MAC      | Implementation     |
   | Value        | Algorithm                     | Requirements       |
   +--------------+-------------------------------+--------------------+
   | HS256        | HMAC using SHA-256            | Required           |
   | HS384        | HMAC using SHA-384            | Optional           |
   | HS512        | HMAC using SHA-512            | Optional           |
   | RS256        | RSASSA-PKCS1-v1_5 using       | Recommended        |
   |              | SHA-256                       |                    |
   | RS384        | RSASSA-PKCS1-v1_5 using       | Optional           |
   |              | SHA-384                       |                    |
   | RS512        | RSASSA-PKCS1-v1_5 using       | Optional           |
   |              | SHA-512                       |                    |
   | ES256        | ECDSA using P-256 and SHA-256 | Recommended+       |
   | ES384        | ECDSA using P-384 and SHA-384 | Optional           |
   | ES512        | ECDSA using P-521 and SHA-512 | Optional           |
   | PS256        | RSASSA-PSS using SHA-256 and  | Optional           |
   |              | MGF1 with SHA-256             |                    |
   | PS384        | RSASSA-PSS using SHA-384 and  | Optional           |
   |              | MGF1 with SHA-384             |                    |
   | PS512        | RSASSA-PSS using SHA-512 and  | Optional           |
   |              | MGF1 with SHA-512             |                    |
   | none         | No digital signature or MAC   | Optional           |
   |              | performed                     |                    |
   +--------------+-------------------------------+--------------------+
```

But after some time, I noticed the most essential word is `asymmetric`:

> Comma-separated list of allowed JOSE **asymmetric** signing algorithms

#### What you expected to happen:

The documentation to be improved to explicitly list the supported algorithms instead of referring to the RFC document that lists much more algorithms than the actual supported ones.

#### How to reproduce it (as minimally and precisely as possible):
Run the kube-apiserver with flag `--oidc-signing-algs=HS256`

#### Anything else we need to know?:

I think the really supported algorithms are explicitly listed here https://github.com/kubernetes/kubernetes/blob/5be21c50c269fc1d28e0bd31ab9dcb572ae7fac5/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go#L225-L237

#### Environment:
- Kubernetes version (use `kubectl version`): 
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by Shubham82** [comment]
Hi @enj @vpnachev

IMO We can do two things here either we will mention the supported signing algorithm [link](https://github.com/kubernetes/kubernetes/blob/5be21c50c269fc1d28e0bd31ab9dcb572ae7fac5/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go#L225-L237) for oidc instead of [RFC document](https://tools.ietf.org/html/rfc7518#section-3.1)  or explicitly mention the supported signing algorithm in the docs. 

What do you suggest?

I'm up for it if nobody is doing that.

**Comment by Shubham82** [comment]
Fixed it here: #105067
